src/main/java/randoop/generation/GrtImpurity.java

```diff
@@ -0,0 +1,513 @@
+    } else {
+      throw new IllegalArgumentException("Unexpected primitive type: " + cls.getName());
+    }
+
+    return methodList;
+  }
+
+  /**
+   * Create and append all statements needed for the String fuzzing operation to the given sequence.
+   * Precondition: Length of the input String to be fuzzed is not 0.
+   *
+   * @param sequence the (non-empty) sequence to append the String fuzzing operation inputs to
+   * @param operation the String fuzzing operation to perform
+   * @param fuzzStatementOffset the offset counter for the number of fuzzing statements added
+   * @return a sequence with the String fuzzing operation inputs appended at the end
+   */
+  private static Sequence appendStringFuzzingInputs(
+      Sequence sequence, StringFuzzingOperation operation, FuzzStatementOffset fuzzStatementOffset)
+      throws NoSuchMethodException {
+    Object stringValue = getStringValue(sequence);
+    int stringLength = stringValue.toString().length();
+
+    if (stringLength == 0 && operation != StringFuzzingOperation.INSERT) {
+      return sequence; // Cannot remove/replace/substring an empty string
+    }
+
+    sequence = appendStringBuilder(sequence, fuzzStatementOffset);
+
+    List<Sequence> fuzzingSequenceList = getStringFuzzingInputs(operation, stringLength);
+    fuzzingSequenceList.add(0, sequence); // prepend
+
+    return Sequence.concatenate(fuzzingSequenceList);
+  }
+
+  /**
+   * Append a StringBuilder constructor statement to the given sequence. The string value used by
+   * the StringBuilder constructor is from the last statement in the sequence.
+   *
+   * @param sequence the sequence to append the StringBuilder constructor to
+   * @param fuzzStatementOffset the offset counter for the number of fuzzing statements added
+   * @return a sequence with the StringBuilder constructor appended at the end
+   */
+  private static Sequence appendStringBuilder(
+      Sequence sequence, FuzzStatementOffset fuzzStatementOffset) throws NoSuchMethodException {
+    Constructor<?> stringBuilderConstructor = StringBuilder.class.getConstructor(String.class);
+    return appendFuzzingOperation(
+        sequence,
+        stringBuilderConstructor,
+        getOutputType(stringBuilderConstructor),
+        fuzzStatementOffset);
+  }
+
+  /**
+   * Get the String value from the given sequence. Precondition: The String value is the last
+   * statement in the sequence.
+   *
+   * @param sequence the sequence to get the String value from
+   * @return the String value from the given sequence
+   * @throws IllegalArgumentException if the String value cannot be obtained
+   */
+  private static Object getStringValue(Sequence sequence) {
+    try {
+      return sequence.getStatement(sequence.size() - 1).getValue();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+          "Invalid sequence, last statement does not have a String value");
+    }
+  }
+
+  /**
```

Comment:
Make the next 6 methods after this one be instance methods in enum values.  This method can probably be dispensed with, except as an abstract method on the enum class itself.